try():
	difficult_function()
expect CallError as exc:
	raise DataCalculatingError(str(exc))